### PR TITLE
Reduce RealNumberQ and refactor

### PR DESCRIPTION
**Pull request by rocky**
Supersedes #820. 

As is common, the changes needed are far too great to do in one programming session. So this is the first part.

### CODE REVIEW

File: mathics/builtin/list/constructing.py [hunk 1]
```diff
@@ -221,33 +227,51 @@ class Range(Builtin):
...
     }
 
     summary_text = "form a list from a range of numbers or other objects"
 
     def eval(self, imin, imax, di, evaluation: Evaluation):
-        "Range[imin_?RealNumberQ, imax_?RealNumberQ, di_?RealNumberQ]"
+        "Range[imin_, imax_, di_]"
+
+        for arg in imin, imax, di:
+            if not is_number(arg):
```
[the remark this answers is not in the document]
Comment (mmatera) [reply mathics/builtin/list/constructing.py:262]: Meaning Real, integer or rational

File: mathics/eval/numbers.py [hunk 2]
```diff
@@ -131,6 +138,13 @@ def cancel(expr):
             return expr
 
 
+def is_number(expr) -> bool:
```
Comment (mmatera) [inline mathics/eval/numbers.py:141]: `is_real_number` maybe?

Comment (rocky) [reply mathics/eval/numbers.py:141]: I have changed this to `is_integer_rational_or_real()` which is unambiguous.

File: mathics/builtin/list/constructing.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rocky) [reply mathics/builtin/list/constructing.py]: The function was poorly (ambiguously or incorrectly) name. That has been corrected. However what it _does_ I think is the correct thing. I compared with WMA.

[the remark this answers is not in the document]
Comment (mmatera) [reply mathics/builtin/list/constructing.py]: I agree. The comment was about the name of the internal function

**Review by mmatera — APPROVED** [review]
LGTM

**Comment by rocky** [comment]
There will be more PRs to remove RealNumberQ. This is just the first.

**Pull request merged**